lib/kube/proxy/sess.go

```diff
@@ -1528,3 +1529,40 @@ func (s *session) retrieveEphemeralContainerCommand(ctx context.Context, usernam
 	}
 	return nil
 }
+
+// concurrentWaitGroup is a wait group that can be used concurrently.
+// It exits when the count reaches 0 for the first time.
+type concurrentWaitGroup struct {
+	cond  *sync.Cond
+	count int
+}
+
+func newConcurrentWaitGroup() concurrentWaitGroup {
+	return concurrentWaitGroup{
+		cond:  sync.NewCond(&sync.Mutex{}),
+		count: 0,
+	}
+}
```

Comment:
Nothing prevents this from being copied, and there's no reason to have 3 separate allocations for this; with some trickery it's also possible to require no initialization.

```suggestion
type concurrentWaitGroup struct {
	cond  sync.Cond
	mu    sync.Mutex
	count int
}
```